lib/src/pubspec.dart

```diff
@@ -30,13 +43,34 @@ class Pubspec {
 
   bool get dependsOnFlutterSdk => dependentSdks.contains('flutter');
 
-  bool get dependsOnAnyVersion {
-    if (dependencies == null) return false;
-    return dependencies.values.any((v) {
-      if (v is String) return v == 'any';
-      if (v is Map) return v['version'] == 'any';
-      return false;
-    });
+  List<String> get tooBroadDependencies {
+    final set = new Set<String>();
+    void inspectDependency(String pkg, dynamic v) {
+      if (v == null || v == 'any') {
+        set.add(pkg);
+        return;
+      }
+      if (v is Map && v.containsKey('version') && !v.containsKey('sdk')) {
+        final version = v['version'];
+        if (version == null || version == 'any') {
+          set.add(pkg);
+          return;
+        }
+      }
+
+      final constraint = v is Map ? v['version'] : v;
+      if (constraint is String) {
+        final vc = new VersionConstraint.parse(constraint);
+        if (vc.allowsAny(_tooLargeVersion)) {
```

Comment:
`vc` is either `VersionRange` or `Version` – I'd just special-case these and avoid the `_tooLargeVersion` variable.